blockindex/contractstaking/indexer.go

```diff
@@ -144,7 +147,7 @@ func (s *Indexer) PutBlock(ctx context.Context, blk *block.Block) error {
 		return nil
 	}
 	// new event handler for this block
-	handler := newContractStakingEventHandler(s.cache, blk.Height())
+	handler := newContractStakingEventHandler(s.cache)
```

Comment:
before this, should also check incoming block height > s.height
```
if blk.Height() <= s.height {
    return nil
}
```

Reply:
fixed